homeassistant/components/webostv/__init__.py

```diff
@@ -1 +1,205 @@
 """Support for WebOS TV."""
+import asyncio
+import logging
+
+from aiopylgtv import PyLGTVCmdException, PyLGTVPairException, WebOsClient
+import voluptuous as vol
+from websockets.exceptions import ConnectionClosed
+
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    CONF_CUSTOMIZE,
+    CONF_FILENAME,
+    CONF_HOST,
+    CONF_ICON,
+    CONF_NAME,
+    ENTITY_MATCH_ALL,
+    EVENT_HOMEASSISTANT_STOP,
+)
+import homeassistant.helpers.config_validation as cv
+
+DOMAIN = "webostv"
+
+CONF_SOURCES = "sources"
+CONF_ON_ACTION = "turn_on_action"
+CONF_STANDBY_CONNECTION = "standby_connection"
+DEFAULT_NAME = "LG webOS Smart TV"
+WEBOSTV_CONFIG_FILE = "webostv.conf"
+
+SERVICE_BUTTON = "button"
+ATTR_BUTTON = "button"
+
+SERVICE_COMMAND = "command"
+ATTR_COMMAND = "command"
+
+CUSTOMIZE_SCHEMA = vol.Schema(
+    {vol.Optional(CONF_SOURCES): vol.All(cv.ensure_list, [cv.string])}
```

Comment:
This config option is intended to limit the available sources in home assistant to a subset of those available from the TV.  (The client was always able to read the full list from the TV, and by default if an explicit source list is not provided, this is what is used.)

Reply:
Oh, ok. Do we really need to limit the list of sources? How long is it usually?

Reply:
It can be very long because it is the concatenation of the available inputs with all installed apps on the TV, including a long list of preinstalled apps which cannot be removed.  (It wasn't me who implemented this, but I guess this was the motivation.)

Reply:
Ok, let's keep it then.